mantidproject/mantid
# Create Reflectometry ConvertSingleSpectrumLambdaToQ child algorithm

**Description of work.**
The existing ISIS reflectometry `convertToQ` routine has been separated into its own child algorithm, as per #23012.

**MERGE AFTER #28566**

**To test:**
Try running the algorithm on single spectrum lambda workspaces and verify the output. Also attempt to use multi-spectra lambda and single spectrum momentum workspaces - you should not be able to and it should leave them unchanged.

Partially resolves #23012 

*This does not require release notes* because **it is an internal change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
